DAO.sol

```diff
@@ -826,32 +947,24 @@ contract DAO is DAOInterface, Token, TokenCreation {
 
 
     function changeAllowedRecipients(address _recipient, bool _allowed) noEther external returns (bool _success) {
-        if (msg.sender != curator)
+        // only allow the curator and the DAO itself (only removing addresses) to make changes to the whitelist
+        if (msg.sender != curator && (msg.sender != address(this) || _allowed))
```

Comment:
`msg.sender != address(this)` is a pointless check. If `msg.sender == curator` we know `msg.seder != address(this)` maybe it should be `_recipient` to disallow removal of the dao from whitelist.